nextroute/schema/fleet.go

```diff
@@ -0,0 +1,234 @@
+type FleetStopDefaults struct {
+	UnassignedPenalty       *int         `json:"unassigned_penalty,omitempty"`
+	Quantity                any          `json:"quantity,omitempty"`
+	HardWindow              *[]time.Time `json:"hard_window,omitempty"`
+	MaxWait                 *int         `json:"max_wait,omitempty"`
+	StopDuration            *int         `json:"stop_duration,omitempty"`
+	TargetTime              *time.Time   `json:"target_time,omitempty"`
+	EarlinessPenalty        *float64     `json:"earliness_penalty,omitempty"`
+	LatenessPenalty         *float64     `json:"lateness_penalty,omitempty"`
+	CompatibilityAttributes *[]string    `json:"compatibility_attributes,omitempty"`
+}
+
+// FleetVehicle holds the fleet input vehicle data.
+type FleetVehicle struct {
+	ID                      string     `json:"id,omitempty"`
+	Start                   *Location  `json:"start,omitempty"`
+	End                     *Location  `json:"end,omitempty"`
+	Speed                   *float64   `json:"speed,omitempty"`
+	Capacity                any        `json:"capacity,omitempty"`
+	ShiftStart              *time.Time `json:"shift_start,omitempty"`
+	ShiftEnd                *time.Time `json:"shift_end,omitempty"`
+	CompatibilityAttributes []string   `json:"compatibility_attributes,omitempty"`
+	MaxStops                *int       `json:"max_stops,omitempty"`
+	MaxDistance             *int       `json:"max_distance,omitempty"`
+	MaxDuration             *int       `json:"max_duration,omitempty"`
+	StopDurationMultiplier  *float64   `json:"stop_duration_multiplier,omitempty"`
+	Backlog                 []string   `json:"backlog,omitempty"`
+	AlternateStops          []string   `json:"alternate_stops,omitempty"`
+	InitializationCost      int        `json:"initialization_cost,omitempty"`
+}
+
+// FleetStop holds the fleet input stop data.
+type FleetStop struct {
+	ID                      string       `json:"id,omitempty"`
+	Position                Location     `json:"position,omitempty"`
+	UnassignedPenalty       *int         `json:"unassigned_penalty,omitempty"`
+	Quantity                any          `json:"quantity,omitempty"`
+	Precedes                any          `json:"precedes,omitempty"`
+	Succeeds                any          `json:"succeeds,omitempty"`
+	HardWindow              *[]time.Time `json:"hard_window,omitempty"`
+	MaxWait                 *int         `json:"max_wait,omitempty"`
+	StopDuration            *int         `json:"stop_duration,omitempty"`
+	TargetTime              *time.Time   `json:"target_time,omitempty"`
+	EarlinessPenalty        *float64     `json:"earliness_penalty,omitempty"`
+	LatenessPenalty         *float64     `json:"lateness_penalty,omitempty"`
+	CompatibilityAttributes *[]string    `json:"compatibility_attributes,omitempty"`
+}
+
+// FleetToNextRoute takes a legacy cloud fleet input and converts it into
+// nextroute input format.
+func FleetToNextRoute(fleetInput FleetInput) (Input, error) {
```

Comment:
```suggestion
func (fleetInput FleetInput) ToNextRoute() (Input, error) {
```

It makes sense to implement the function on the struct directly, rather than passing the schema as an argument.

Reply:
sure